x-pack/plugins/ml/public/components/field_type_icon/field_type_icon.js

```diff
@@ -11,86 +11,57 @@ import { EuiToolTip } from '@elastic/eui';
 
 // don't use something like plugins/ml/../common
 // because it won't work with the jest tests
-import { ML_JOB_FIELD_TYPES } from '../../../common/constants/field_types';
-import { FormattedMessage, injectI18n } from '@kbn/i18n/react';
+import { getMLJobTypeAriaLabel } from '../../util/field_types_utils';
+import { i18n } from '@kbn/i18n';
 
-export const FieldTypeIcon = injectI18n(function FieldTypeIcon({ tooltipEnabled = false, type, intl }) {
-  let ariaLabel = '';
-  let iconClass = '';
+export const FieldTypeIcon = ({ tooltipEnabled = false, type, needsAria = true }) => {
+  const ariaLabel = getMLJobTypeAriaLabel(type);
+
+  if (!ariaLabel) {
+    // if type doesn't match one of ML_JOB_FIELD_TYPES
+    // don't render the component at all
+    return null;
+  }
+
+  const iconClass = ['field-type-icon'];
   let iconChar = '';
 
   switch (type) {
-    case ML_JOB_FIELD_TYPES.BOOLEAN:
-      ariaLabel = intl.formatMessage({
-        id: 'xpack.ml.fieldTypeIcon.booleanTypeAriaLabel',
-        defaultMessage: 'boolean type'
-      });
-      iconClass = 'fa-adjust';
+
+    // icon class names
+    case 'boolean':
```

Comment:
is there a reason why these have been changed to literals?

Reply:
@jgowdyelastic, the idea was to move the processing ML job types out from icon component.
Since the type (passing into that component) is already synchronized with ML types as well as checked whether the pass type values are valid, why do we need to check? -- I thought.

Now I understand, that since we have a single data point, once types may be changed somewhere, my code likely to be broken.
It needs to put ml job types constants back.

However, I wonder, -- how predictable is that for a person to see that the decision to render or not to render this icon component relies on the fact that there is an ariaLabel or there is not. (21st line).

From another perspective, @peteharverson advised me that it would be better to keep `field_types_utils.js` as simple as possible and remove checking type existence from there.
At the same time, I need to store that functionality somewhere, as I need to use those aria labels at 3 different places.

In the meanwhile, I left a comment at `field_type_icon.js` to explain, why it relies on ariaLabel existence.